### DISCUSSION TITLE
Idea: Gang wide messaging

### DISCUSSION THREAD

**Discussion by lee8oi** [DexterHuang/CyberCodeOnline#2242] [Ideas]
I believe it would be good to add a new message dialog that can be configured for gang wide messaging. I think it would be useful for permitted gang members to have a means of sending a message out to all members of their gang. Useful for announcements, reminders, etc.

This should include adding a permission that can be enabled on select members of the gang in order to restrict access to this ability. Keeping it limited to trusted players to avoid abuse or excessive use.

A mail button could be added to gang list, headquarters, or both. Maybe a simple @gang tag for recipient box that would mark the players own gang members as recipients.

A form of this dialog could also benefit players in general by allowing them to selectively message multiple players at once but some players may abuse it to message many players at once. So certain restrictions maybe needed to deter abuse. Such as restricting how often messages can be sent this way, how many players can be tagged at once, even the level or status players must achieve to enable it.

**Comment by S0M3-DUD3** [comment]
That's a great idea. I could msg everybody the same msg as gang leader to start or inform my member about events I'm planing or that running currently. That would Safe so much time over msg everybody on there own and copy-paste the msg.

**Reply by lee8oi** [reply]
Exactly. I think managing a gang does include the ability to inform all members of certain events, tasks, etc. Visiting each members profile to send a message is tedious.

**Comment by GhostMan612** [comment]
Who writes the code for the branches or forks? 

How do we get the developers to respond or could we do it ourselves as a submit and wait for the commit & push...

I'm sure that since they put the game code on GitHub, someone can probably pull the git and edit or add things and side load to their own device. Although that would probably ban them from using the game online you could still play it offline and custom build it to your liking. And if you build it good enough, the developers may take you up on the new code and decide to push it the public...

It's all a matter of getting android studio installed and configured correctly with all the right plugins and then have the knowledge of which coding was used, I see react.js but that's the dashboards and UI. Kotlin / Java is the underlying code...

I say, suggest it to the developers with a sample code and see if they implement it...


In My app I have a sovereign communications module that uses the same premise of public/group/private(encrypted) chats and photo sharing and push-to-talk which uses a passphrase for groups/families. The part I am working on now is getting it to mass text the entire group/family without having to add everyone as a contact, although the sovereign mesh is your contact list it should still be able to just mass text the group in one swoop.
